enforcer/datapath_proxy.go

```diff
@@ -0,0 +1,543 @@
+			net.IPv4(remote.Addr[0], remote.Addr[1], remote.Addr[2], remote.Addr[3]).String(),
+			syscall.IPPROTO_TCP,
+			uint16(local.Port),
+			uint16(remote.Port),
+			constants.DefaultConnMark,
+		); connterror != nil {
+			zap.L().Error("Unable to mark flow")
+		}
+
+	}
+
+	return fd, nil
+}
+
+//CompleteEndPointAuthorization -- Aporeto Handshake on top of a completed connection
+//We will define states here equivalent to SYN_SENT AND SYN_RECEIVED
+func (p *Proxy) CompleteEndPointAuthorization(backendip string, backendport uint16, upConn net.Conn, downConn int, contextID string) error {
+	puContext, err := p.datapath.contextTracker.Get(contextID)
+	if err != nil {
+		zap.L().Error("Did not find context")
+	}
+	puContext.(*PUContext).Lock()
+	defer puContext.(*PUContext).Unlock()
+	pu := puContext.(*PUContext)
+	//addr := upConn.RemoteAddr().String()
+
+	if pu.PUType == constants.LinuxProcessPU {
+		//Are we client or server proxy
+
+		if len(puContext.(*PUContext).Ports) > 0 && puContext.(*PUContext).Ports[0] != "0" {
+
+			return p.StartServerAuthStateMachine(backendip, backendport, upConn, downConn, contextID)
+		}
+		//We are client no advertised port
+		return p.StartClientAuthStateMachine(backendip, backendport, upConn, downConn, contextID)
+
+	}
+	//Assumption within a container two applications talking to each other won't be proxied.
+	//If backend ip is non local we are client else we are server
+	islocalIP := func() bool {
+		for _, ip := range p.IPList {
+			if ip == backendip {
+				return true
+			}
+		}
+		return false
+	}()
+	if islocalIP {
+		return p.StartServerAuthStateMachine(backendip, backendport, upConn, downConn, contextID)
+	}
+	return p.StartClientAuthStateMachine(backendip, backendport, upConn, downConn, contextID)
+
+}
+
+//getProxyPort for a given PU
+// func (p *Proxy) getProxyPort(puInfo *policy.PUInfo) string {
+// 	port, ok := puInfo.Runtime.Options().Get("proxyPort")
+// 	if !ok {
+// 		port = constants.DefaultProxyPort
+// 	}
+// 	return port
+// }
+
+//StartClientAuthStateMachine -- Starts the aporeto handshake for client application
+func (p *Proxy) StartClientAuthStateMachine(backendip string, backendport uint16, upConn net.Conn, downConn int, contextID string) error {
+	//We are running on top of TCP nothing should be lost or come out of order makes the state machines easy....
+	puContext, err := p.datapath.contextTracker.Get(contextID)
+	if err != nil {
+		zap.L().Error("Did not find context")
+	}
+	conn := NewTCPConnection()
+
+	toAddr, err := syscall.Getpeername(downConn)
+	//fmt.Println(toAddr.(*syscall.SockaddrInet4).)
+	// ipv4addr := toAddr.(*syscall.SockaddrInet4).Addr
+	// port := toAddr.(*syscall.SockaddrInet4).Port
+	if err != nil {
+		zap.L().Error("Peer Name Failed", zap.Error(err))
+	}
+
+L:
+	for conn.GetState() == TCPSynSend {
+		msg := make([]byte, 1024)
+		for {
+			switch conn.GetState() {
+			case TCPSynSend:
```

Comment:
These are not really Syn/SynACK packets, but rather authorization packets over an estalblished connection. Let's change the constants to reflect this. Its confusing otherwise.